File: java-library/src/main/java/com/gblninja/tag/type/GblProgLz4.java
package com.gblninja.tag.type;


import com.gblninja.tag.GblType;
import com.gblninja.tag.Tag;
import com.gblninja.tag.TagHeader;
import com.gblninja.tag.TagWithHeader;

public class GblProgLz4 implements Tag, TagWithHeader {
    private final TagHeader tagHeader;
    private final GblType tagType;
    private final byte[] tagData;

    public GblProgLz4(TagHeader tagHeader, GblType tagType, byte[] tagData) {
        this.tagHeader = tagHeader;
        this.tagType = tagType;
        this.tagData = tagData;
    }

    @Override
    public GblType getTagType() {
        return tagType;
    }

    @Override
    public Tag copy() {
        return new GblProgLz4(tagHeader, tagType, new byte[0]);
    }

    @Override
    public TagHeader getTagHeader() {
        return tagHeader;
    }

    @Override
    public byte[] getTagData() {
        return tagData;
    }
}

File: gbl/src/main/java/com/onix/gbl/parser/type/GblProgLz4.kt
package parser.data.type

import parser.data.tag.TagHeader
import parser.data.tag.TagInterface

data class GblProgLz4(
    override val tagHeader: TagHeader,
    override val tagType: GblType,
    override val tagData: ByteArray
): TagInterface {
    override fun copy(): TagInterface {
        return GblProgLz4(
            tagHeader = tagHeader,
            tagType = tagType,
            tagData = arrayOf<Byte>().toByteArray()
        )
    }
}
